doaj/software-versions.yml

```diff
@@ -9,117 +9,200 @@
-      changed_when: false
-      ignore_errors: true
-      args:
-        warn: no
+    - name: Set Libxml version message
+      set_fact:
+        libxml_msg: "Libxml Version: {{ libxml_version.stdout }}"
+      when: libxml_version is defined and 'stdout' in libxml_version
+
+    - name: Collect all messages
+      set_fact:
+        messages: "{{ [ubuntu_msg|default(''), pyhon_msg|default(''), java_msg|default(''), metricbeat_msg|default(''), elasticsearch_msg|default(''),
+        local_elasticsearch_msg|default(''), kibana_msg|default(''), logstash_msg|default(''), redis_msg|default(''), kafka_msg|default(''),
+        nginx_msg|default(''), debian_iso_list_msg|default(''), ansible_msg|default(''), libxml_msg|default('')] | select('ne', '') | list }}"
 
     - name: Display gathered versions
       debug:
-        msg:
-          - "Ubuntu Version: {{ ubuntu_version.stdout if (ubuntu_version is defined and 'stdout' in ubuntu_version) else 'N/A' }}"
-          - "Python Version: {{ python_version.stdout if (python_version is defined and 'stdout' in python_version) else 'N/A' }}"
-          - "Elasticsearch Version: {{ elasticsearch_version.json.version.number if (elasticsearch_version is defined and 'json' in elasticsearch_version) else 'N/A' }}"
-          - "Kibana Version: {{ kibana_version.json.version.number if (kibana_version is defined and 'json' in kibana_version) else 'N/A' }}"
-          - "Logstash Version: {{ logstash_version.stdout if (logstash_version is defined and 'stdout' in logstash_version) else 'N/A' }}"
-          - "Metricbeat Version: {{ metricbeat_version.stdout if (metricbeat_version is defined and 'stdout' in metricbeat_version) else 'N/A' }}"
-          - "Redis Version: {{ redis_version.stdout if (redis_version is defined and 'stdout' in redis_version) else 'N/A' }}"
-          - "Kafka Version: {{ kafka_version.stdout if (kafka_version is defined and 'stdout' in kafka_version) else 'N/A' }}"
-          - "nginx Version: {{ nginx_version.stderr if (nginx_version is defined and 'stdout' in nginx_version) else 'N/A' }}" # Note: nginx -v writes version to stderr
-          - "Debian Version: {{ (debian_version['content'] | b64decode).strip() if (debian_version is defined) else 'N/A' }}"
-          - "Ansible Version: {{ ansible_version.stdout if (ansible_version is defined and 'stdout' in ansible_version) else 'N/A' }}" # Extract the version number
-          - "Ansible Version currently used: {{ ansible_version.full }}"
-          - "LibXML Version: {{ libxml_version.stdout if (libxml_version is defined and 'stdout' in libxml_version) else 'N/A' }}"
-          - "Java Version: {{ java_version.stderr if (java_version is defined and 'stdout' in java_version) else 'N/A' }}" # Note: java -version writes version to stderr
-      tags:
-        - output
\ No newline at end of file
+        msg: "{{ messages }}"
+
+- hosts: localhost
+  vars:
+    pycountry_dir: "{{ ansible_env.PWD }}/pycountry"
```

Comment:
is this better in `/tmp/`? or in userspace as it is here

Reply:
I used ansible_env.PWD so that this script can work on any OS. We can move to tmp/ if that is better. We are deleting the directory after the task complete.

Reply:
I'm happy with using the local environment if it clears up, thanks